app-modules/service-management/src/Jobs/ServiceMonitoringCheckJob.php

```diff
@@ -0,0 +1,104 @@
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    <https://www.canyongbs.com> or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace AidingApp\ServiceManagement\Jobs;
+
+use AidingApp\ServiceManagement\Enums\ServiceMonitoringFrequency;
+use AidingApp\ServiceManagement\Models\ServiceMonitoringTarget;
+use AidingApp\ServiceManagement\Notifications\ServiceMonitoringNotification;
+use App\Models\User;
+use Illuminate\Contracts\Queue\ShouldBeUnique;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Foundation\Queue\Queueable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Support\Facades\Http;
+use Illuminate\Support\Facades\Notification;
+
+class ServiceMonitoringCheckJob implements ShouldQueue, ShouldBeUnique
+{
+    use Dispatchable;
+    use InteractsWithQueue;
+    use Queueable;
+    use SerializesModels;
+
+    public function __construct(public ServiceMonitoringTarget $serviceMonitoringTarget) {}
+
+    public function uniqueId(): string
+    {
+        return $this->serviceMonitoringTarget->getKey();
+    }
+
+    /**
+     * Return the period for which this job should be unique for, its interval plus half an hour, in seconds
+     */
+    public function uniqueFor(): int
+    {
+        $seconds = match ($this->serviceMonitoringTarget->frequency) {
+            ServiceMonitoringFrequency::OneHour => 60 * 60,
+            ServiceMonitoringFrequency::TwentyFourHours => 24 * 60 * 60,
+        };
+
+        return $seconds + (30 * 60);
+    }
+
+    public function handle(): void
+    {
+        $response = Http::get($this->serviceMonitoringTarget->domain);
+
+        $this->serviceMonitoringTarget->history()->create([
+            'response' => $response->status(),
+            'response_time' => $response->transferStats->getTransferTime() ?? 0,
+            'succeeded' => $response->status() === 200,
+        ]);
+
+        if ($response->status() !== 200) {
+            /** @var Collection<int, User> $recipients */
+            $recipients = $this->serviceMonitoringTarget->users()->get();
+
+            $this->serviceMonitoringTarget->teams()->each(function ($team) use ($recipients) {
+                /** @var Collection<int, User> $users */
+                $users = $team->users()->get();
+
+                $recipients->concat($users)->unique();
+            });
+
+            Notification::send($recipients, new ServiceMonitoringNotification($this->serviceMonitoringTarget->history));
```

Comment:
Like I said in the other comment, we want to change the history relationship to be a one to many.

So `$this->serviceMonitoringTarget->history` would not work here.

You can save the history record to a variable on like `84` here and dispatch using that record specifically.